synapse/http/client.py

```diff
@@ -81,13 +82,30 @@ def __init__(self, hs):
         )
         self.user_agent = hs.version_string
         self.clock = hs.get_clock()
+        self.reactor = hs.get_reactor()
         if hs.config.user_agent_suffix:
             self.user_agent = "%s %s" % (self.user_agent, hs.config.user_agent_suffix,)
 
         self.user_agent = self.user_agent.encode('ascii')
 
     @defer.inlineCallbacks
     def request(self, method, uri, data=b'', headers=None):
+
+        # Check our IP whitelists/blacklists before making the request.
+        if self.blacklist:
+            split_uri = URI.fromBytes(uri.encode('utf8'))
+            address = yield self.reactor.resolve(split_uri.host)
```

Comment:
errrm surely this has a massive security hole, because there is no guarantee that we will connect to the same IP address when we come to do the fetch?

Reply:
This isn't new code, this is existing code that's moved. And yes, potentially.

Reply:
Oh, hmm, I guess it does add something, as it was in `connect` before... let me think on this.

Reply:
Actually, looking at it, the current code has the same problem (since it then passes it to HostnameEndpoint for IPv6).